client/src/components/ProductCard.tsx

```diff
@@ -143,6 +145,31 @@ function ProductCard ({ product, bookmark, log }: Props) {
     }
   };
 
+  const handleShareClick: React.MouseEventHandler<HTMLDivElement> = async () => {
+    const projectURL = 'https://oriori.fly.dev/home';
+    try{
+        if(navigator.share){
+          await navigator.share({
+            title: `${product.product_name}`,
+            text: `${product.id}:${product.product_name}`,
+            url: `${projectURL}#${product.id}`,
+          })
+        } else {
+          const shareURL = `${projectURL}#${product.id}`;
+            await navigator.clipboard.writeText(shareURL);
+            console.log("💌",product.id, ":", product.product_name,":", shareURL);
+        }
+        setHasShared(true);
+        setTimeout(() => {
+          setHasShared(false);
+        }, 3000);
+      } catch (error){
+        console.error("💩",error);
+    }
+  } 
+  
+
+
   return (
     <div className='product__card'>
```

Comment:
The share button works perfectly, including the automatic copy to the clipboard! I noticed that the cards don't seem to have id's on them so that, for example, when visiting `oriori.fly.dev/home#3` I'm taken to the product card element that looks like:
```html
<div className="product__card" id="3">...</div>
```
But I may have missed the "id" tag somewhere else. Has that been added yet? If not, please make sure that each card includes an "id" tag with the product id so that the links work correctly :)

Reply:
For this last part, in theory, you needed the following in order to run:
```html
<div className="product__card: id={`${product.id}`}>
<a href={`#${product.id}`}>{product.id}</a>...</div>
```
Enlight me if I am incorrect, I think the `<a href="#{product.id}">{product.id}</a>` are necessary for the anchor to work properly.

Reply:
This works now! Thank you! When I visit `.../home#3` it takes me to the card that renders with `id="3"`. 🥳